File: Sorting/Python/Find H-Index.py
class Solution:
    def hIndex(self, citations):
        #code here
        n = len(citations)
        freq = [0] * (n + 1)
        
        for cit in citations:
            if cit > n:
                freq[n] += 1
            else:
                freq[cit] += 1
        
        idx = n
        s = freq[n]
        
        while s < idx:
            idx -= 1
            s += freq[idx]
        return idx

File: Sorting/C++/Find H-Index.cpp
class Solution {
  public:
    int hIndex(vector<int>& citations) {
        // code here
        int n = citations.size();
        vector<int> freq(n + 1);
        
        for (int cit: citations) {
            if (cit > n) {
                freq[n]++;
            }
            else {
                freq[cit]++;
            }
        }
        
        int idx = n;
        int s = freq[n];
        
        while (s < idx) {
            idx--;
            s += freq[idx];
        }
        return idx;
    }
};
